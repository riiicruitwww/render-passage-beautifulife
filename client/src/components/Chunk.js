import React, { Fragment } from 'react';
import './Chunk.scss';

const Chunk = ({ chunk, chunkMap }) => {
  const renderChunk = () => {
    const mappedChunk = chunkMap[chunk.chunk_id];
    const chunkText = mappedChunk[chunk.type];
    let stringArray = [chunkText];
    let order;

    if (chunk.children.length) {
      chunk.children.forEach(item => {
        if (item.data.styles[0] === 'blank') {
          const { begin, offset } = item.data;
          order = `----${item.order}----`;
          stringArray = [
            chunkText.substring(0, begin),
            { order },
            chunkText.substring(begin + offset)
          ];
        }
      });
    }

    return stringArray.map((string, index) => {
      if (string.order) {
        return (
          <span key={index} className="Chunk order">
            {order}
          </span>
        );
      }

      return (
        <span key={index} className="Chunk" data-id={chunk.chunk_id}>
          {string}
        </span>
      );
    });
  };

  return <Fragment>{renderChunk()}</Fragment>;
};

export default Chunk;
