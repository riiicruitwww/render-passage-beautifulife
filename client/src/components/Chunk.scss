.Chunk {
  &.order {
    display: inline-block;
  }
}
