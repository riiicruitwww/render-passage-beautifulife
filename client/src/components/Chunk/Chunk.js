import React, { Fragment } from 'react';
import './Chunk.scss';

const Chunk = ({ chunk: { children, chunk_id, type }, chunkMap }) => {
  const renderChunk = () => {
    const mappedChunk = chunkMap[chunk_id];
    const chunkText = mappedChunk[type];
    let stringArray = [chunkText];

    if (children.length !== 0) {
      children.forEach(item => {
        if (item.data.styles[0] === 'blank') {
          const { begin, offset } = item.data;
          const order = `----${item.order}----`;
          stringArray = [
            chunkText.substring(0, begin),
            order,
            chunkText.substring(begin + offset)
          ];
        }
      });
    }

    return stringArray.map((string, index) => {
      return (
        <span
          key={index}
          className={`Chunk ${index === 1 ? 'order' : ''}`}
          data-id={chunk_id}
        >
          {string}
        </span>
      );
    });
  };

  return <Fragment>{renderChunk()}</Fragment>;
};

export default Chunk;
