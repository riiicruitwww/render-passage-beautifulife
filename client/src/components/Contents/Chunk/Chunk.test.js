import React from 'react';
import { shallow } from 'enzyme';
import Chunk from './Chunk';

describe('Chunk', () => {
  describe('about render', () => {
    const props = {
      chunk: {
        chunk_id: 'testId',
        type: 'testType',
        children: []
      },
      chunkMap: {
        testId: {
          testType: 'testText'
        }
      }
    };
    const wrapper = shallow(<Chunk {...props} />);

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('component rendered', () => {
      expect(wrapper.find('.Chunk').length).toBe(1);
    });
  });
});
