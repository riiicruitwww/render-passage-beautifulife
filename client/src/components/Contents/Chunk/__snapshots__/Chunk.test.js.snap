// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chunk about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chunk
    chunk={
      Object {
        "children": Array [],
        "chunk_id": "testId",
        "type": "testType",
      }
    }
    chunkMap={
      Object {
        "testId": Object {
          "testType": "testText",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <span
          className="Chunk "
          data-id="testId"
        >
          testText
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "testText",
          "className": "Chunk ",
          "data-id": "testId",
        },
        "ref": null,
        "rendered": "testText",
        "type": "span",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span
            className="Chunk "
            data-id="testId"
          >
            testText
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "testText",
            "className": "Chunk ",
            "data-id": "testId",
          },
          "ref": null,
          "rendered": "testText",
          "type": "span",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
